src/compiler/scala/tools/nsc/typechecker/Macros.scala

```diff
@@ -60,8 +63,24 @@ trait Macros extends MacroRuntimes with Traces with Helpers {
    */
   protected def findMacroClassLoader(): ClassLoader = {
     val classpath = global.classPath.asURLs
-    macroLogVerbose("macro classloader: initializing from -cp: %s".format(classpath))
-    ScalaClassLoader.fromURLs(classpath, self.getClass.getClassLoader)
+    def newLoader = () => {
+      macroLogVerbose("macro classloader: initializing from -cp: %s".format(classpath))
+      ScalaClassLoader.fromURLs(classpath, self.getClass.getClassLoader)
+    }
+
+    import scala.tools.nsc.io.Jar
+    import scala.reflect.io.{AbstractFile, Path}
+    val locations = classpath.map(u => Path(AbstractFile.getURL(u).file))
```

Comment:
Cross-referencing with regression report in https://github.com/scala/scala-dev/issues/480